### PR TITLE
zscaler_zia: fix mapping of user identities

### LINKED ISSUE #9040: zscaler_zia: user identitiies are not properly mapped
In some cases, user identities (`user.id` and `related.user`) are not properly mapped according to ECS guidelines. In the error cases, `user.id` is not retained at all and `related.user` may omit `user.name` values depending on their format.

This problem exists in the dns, firewall and web data streams.

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

See title.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

- Fixes #9040

### CODE REVIEW

File: packages/zscaler_zia/data_stream/web/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -243,6 +247,16 @@ processors:
           value: '{{{_ingest._value}}}'
           allow_duplicates: false
       ignore_missing: true
+  - append:
+      field: related.user
+      value: '{{{user.name}}}'
+      allow_duplicates: false
+      if: ctx.user?.name != null && !(ctx.user.name instanceof List)
```
Comment (chrisberkhout) [inline packages/zscaler_zia/data_stream/web/elasticsearch/ingest_pipeline/default.yml:254]: If it's a list you could do foreach to append each one.

File: packages/zscaler_zia/data_stream/alerts/_dev/test/pipeline/test-alerts.log-expected.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
     "expected": [
         {
-            "@timestamp": "2023-12-31T12:01:04.000Z",
+            "@timestamp": "2024-12-31T12:01:04.000Z",
```
Comment (chrisberkhout) [inline packages/zscaler_zia/data_stream/alerts/_dev/test/pipeline/test-alerts.log-expected.json:4]: I didn't know about the yearless [BSD syslog format](https://datatracker.ietf.org/doc/html/rfc3164#section-4.1.2)  until I saw this.

**Review by chrisberkhout — APPROVED** [review]

File: packages/zscaler_zia/data_stream/alerts/_dev/test/pipeline/test-alerts.log-expected.json (hunk 2 above)
Comment (efd6) [reply packages/zscaler_zia/data_stream/alerts/_dev/test/pipeline/test-alerts.log-expected.json:4]: One year should be enough for anyone.

File: packages/zscaler_zia/data_stream/web/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (efd6) [reply packages/zscaler_zia/data_stream/web/elasticsearch/ingest_pipeline/default.yml:254]: I actually had that code in, but never saw a case in any of the samples I could find. So I removed it. The type check here is fossilised paranoia.

**Pull request merged**